ament_cmake_core/ament_cmake_uninstall_target-extras.cmake

```diff
@@ -33,7 +33,11 @@ if(AMENT_CMAKE_UNINSTALL_TARGET)
 
   # register uninstall target to run generated CMake script
   if(NOT TARGET uninstall)
-    add_custom_target(uninstall
-      COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
+    add_custom_target(uninstall)
   endif()
+  
+  string(RANDOM LENGTH 8 rand)
```

Comment:
Rather than using a random number here I would suggest to use a counter and increment it each time a new uninstall target is created.